VMware.CloudFoundation.PowerManagement.psm1

```diff
@@ -744,6 +744,104 @@ Function Set-VsanClusterPowerStatus {
+
+        $checkServer = (Test-NetConnection -ComputerName $server -Port 443).TcpTestSucceeded
+        if ($checkServer) {
+            Write-PowerManagementLogMessage -Type INFO -Message "Connecting to '$server'..."
+            if ($DefaultVIServers) {
+                Disconnect-VIServer -Server * -Force -Confirm:$false -WarningAction SilentlyContinue  -ErrorAction  SilentlyContinue | Out-Null
+            }
+
+            # Prepare VxRail rest API headers and payload
+            $payloadTest = @{ dryrun = 'true'} | ConvertTo-Json
+            $payloadRun = @{ dryrun = 'false'} | ConvertTo-Json
+            $base64AuthInfo = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes(("{0}:{1}" -f $user, $pass))) # Create Basic Authentication Encoded Credentials
+            $headers = @{"Content-Type" = "application/json" }
+            $headers.Add("Authorization", "Basic $base64AuthInfo")
+			$Uri = "https://$server/rest/vxm/v1/cluster/shutdown"
+
+            Write-PowerManagementLogMessage -Type INFO -Message "Starting VxRail Cluster shutdown dry run." -Colour Yellow
+			$respond = Invoke-WebRequest -Method POST -Uri $uri -Headers $headers -Body $payloadTest -UseBasicParsing
+            if($respond.StatusCode -eq "202" -or $respond.StatusCode -eq "200") {
+                $requestID = $respond.content | ConvertFrom-JSON
+                Write-PowerManagementLogMessage -Type INFO -Message "VxRail Cluster shutdown request accepted(ID:$($requestID.request_id))"
+                $uri2 = "https://$server/rest/vxm/v1/requests/$($requestID.request_id)"
+                $loopCounter = 0
+                while ($loopCounter -lt 13){
+                    $respond2 = Invoke-WebRequest -Method GET -Uri $uri2 -Headers $headers -UseBasicParsing
+                    if ($respond2.StatusCode -eq "202" -or $respond2.StatusCode -eq "200") {
+                        $checkProgress = $respond2.content | ConvertFrom-JSON
+                        if($checkProgress.state -match "COMPLETED" -or $checkProgress.state -match "FAILED" ) {
+                            break
+                        }
+                    }
+                    sleep(10)
+                    $loopCounter += 1
+                }
+                
+                if($checkProgress.extension.passed -match "true") {
+                    Write-PowerManagementLogMessage -Type INFO -Message "VxRail Cluster shutdown dry run: SUCCESSED."
+                    Write-PowerManagementLogMessage -Type INFO -Message "Starting VxRail Cluster shutdown."
+
+                    $respond = Invoke-WebRequest -Method POST -Uri $uri -Headers $headers -Body $payloadRun -UseBasicParsing
+                    if ($respond.StatusCode -eq "202" -or $respond.StatusCode -eq "200") {
+                        return $true
+                    } else {
+                        Write-PowerManagementLogMessage -Type ERROR -Message "Invoke VxRail Cluster shutdown: FAILED" -Colour Red
```

Comment:
```suggestion
                        Write-PowerManagementLogMessage -Type ERROR -Message "VxRail cluster shutdown: FAILED" -Colour Red
```

Reply:
addressed in commit f8c5a69ef84a1a40675a91879ad0a67701fba108